src/core/gui/dialog/SettingsDialog.cpp

```diff
@@ -914,4 +915,5 @@ void SettingsDialog::save() {
 
     this->control->getWindow()->setGtkTouchscreenScrollingForDeviceMapping();
     this->control->initButtonTool();
-}
\ No newline at end of file
+    this->control->getWindow()->getXournal()->onSettingsChanged();
```

Comment:
Not sure whether this is the best solution, but I think this is fine for now. It may be possible that this will lead to deadlocks in the future, but we can revisit this decision if that happens.